src/fork.mjs

```diff
@@ -0,0 +1,53 @@
+import ensureIterable from './internal/ensure-iterable'
+import { Exchange } from './internal/queues'
+
+export default function * fork (iterable) {
+  const iterator = ensureIterable(iterable)[Symbol.iterator]()
+
+  let iterableNumber = 0
+  let noNewIterables = false
+  const exchange = new Exchange()
+  let done = false
+
+  function fetch () {
+    const newItem = iterator.next()
+    if (newItem.done) {
+      done = true
+    } else {
+      exchange.push(newItem.value)
+    }
+  }
+
+  function returnIterator () {
+    if (noNewIterables && iterableNumber === 0) {
+      if (typeof iterator.return === 'function') iterator.return()
+    }
+  }
+
+  function * forkGen (a) {
+    try {
+      while (true) {
+        if (!a.isEmpty()) {
+          yield a.shift()
+        } else if (done) {
+          return
```

Comment:
I believe you need to make sure that you capture the return value of the source iterable here

Reply:
Test case:
```js
const iter = (function * () {
  yield 1;
  return 2;
})()
iter.next() // { done: false, value: 1}
iter.next() // { done: true, value: 2} (I think you're missing the value here)
```

Reply:
Wow, I completely missed that part of the specs (never used). Thanks